insalan/tournament/payment.py

```diff
@@ -0,0 +1,145 @@
+"""Handling of the payment of a registration (player and manager)"""
+
+import logging
+
+from insalan.payment.models import ProductCategory
+from insalan.payment.hooks import PaymentHooks, PaymentCallbackSystem
+
+from django.utils.translation import gettext_lazy as _
+
+from insalan.tickets.models import Ticket
+from insalan.tournament.models import Player, Manager, PaymentStatus
+
+
+logger = logging.getLogger("insalan.tournament.hooks")
+
+
+class PaymentHandler(PaymentHooks):
+    """Handler of the payment of a ticket/registration"""
+
+    @staticmethod
+    def fetch_registration(tourney, user):
+        """
+        Fetch a registration for a user in a tournament.
+
+        Returns a tuple (reg, is_manager), which is pretty explicit.
+        """
+        # Find a registration on that user within the tournament
+        # Could they be player?
+        reg = Player.objects.filter(
+            team__tournament=tourney, user=user, payment_status=PaymentStatus.NOT_PAID
+        )
+        if len(reg) > 1:
+            raise RuntimeError(_("Plusieurs inscription joueur⋅euse à un même tournoi"))
+        if len(reg) == 1:
+            return (reg[0], False)
+
+        reg = Manager.objects.filter(
+            team__tournament=tourney, user=user, payment_status=PaymentStatus.NOT_PAID
+        )
+        if len(reg) > 1:
+            raise RuntimeError(_("Plusieurs inscription manager à un même tournoi"))
+        if len(reg) == 0:
+            raise RuntimeError(
```

Comment:
the translation may fail because of f-string

Reply:
Yup, it will in fact, so I fixed it.